### PR TITLE
StaticLib: Add static library support

### PR DESCRIPTION
**Pull request by 1480c1**
Definitely could use more clean up

### CODE REVIEW

**Comment by tianjunwork** [comment]
Hi @Austin-Hu,  could you review and verify this PR?
FYI: https://github.com/OpenVisualCloud/SVT-HEVC/commit/835ec04a4bc9430169657abce5e41cdc6a73f129
Thank you.

**Comment by tianjunwork** [comment]
Hmm, @1480c1 did you forget the change in build.sh and README.md?

**Comment by 1480c1** [comment]
Haven't added those yet since I'm still testing to make sure if there's any way to clean it up/do it better

File: Build/linux/build.sh [hunk 1]
```diff
@@ -1,79 +1,287 @@
...
-
-    # Compile the App
-    make -j SvtVp9EncApp
-
-    cd ..    
+build() (
+    build_type="Release"
+    while [ -n "$*" ]; do
+        match=$(printf %s "$1" | tr '[:upper:]' '[:lower:]')
+        case "$match" in
```
Comment (1480c1) [inline Build/linux/build.sh]:
```suggestion
        case "$(printf %s "$1" | tr '[:upper:]' '[:lower:]')" in
```

File: Build/linux/build.sh [hunk 2]
```diff
@@ -1,79 +1,287 @@
...
-        -DCMAKE_RANLIB=`which $RANLIB_COMPILER`    \
-
-    # Compile the App
-    make -j SvtVp9EncApp
-
-    cd ..    
+build() (
+    build_type="Release"
+    while [ -n "$*" ]; do
+        match=$(printf %s "$1" | tr '[:upper:]' '[:lower:]')
```
Comment (1480c1) [inline Build/linux/build.sh]:
```suggestion
```

**Pull request merged**